rmf_site_format/src/misc.rs

```diff
@@ -488,14 +522,14 @@ pub struct Pending;
 pub struct Original<T>(pub T);
 
 /// Marks that an entity represents a group
-#[derive(Clone, Copy, Debug, Default)]
+#[derive(Clone, Copy, Debug, Default, PartialEq, Eq)]
```

Comment:
What started as a "Why are we deriving `Eq` for a unit struct" ended up in an interesting journey. I feel this was done to please the compiler for the `skip_serializing_if` function in the [Site](https://github.com/open-rmf/rmf_site/blob/1bb13d0ae0f29c9cbb1ed1caca37dbc1fc2f718b/rmf_site_format/src/site.rs#L146-L147). If the function is changed to just be the same as the rest of the structs in the site `BTreeMap::is_empty` this can be reverted

Reply:
[8c09224](https://github.com/open-rmf/rmf_site/pull/242/commits/8c09224bfc0b6b7aae1831a5a88cd3d1995e505c)

Reply:
I should have brought up, the `model_instances` in Site has the same issue where it uses `is_default` instead of `BTreeMap::is_empty`

Reply:
Oh I attempted to modify that as well, but `model_instances` has a `ModelMarker` component which has the `PartialEq` trait as well. I saw that it was the original design of the struct so I left it as it is. Happy to update that if there wasn't a particular reason it is needed though

Reply:
Yea I wouldn't change existing code (at least to avoid blowing up the diff even more) but would just change https://github.com/open-rmf/rmf_site/blob/c8e1fb233c44dd96aa9e887513075f855a52b244/rmf_site_format/src/site.rs#L149

to also use `BTreeMap::is_empty`

Reply:
[19483c4](https://github.com/open-rmf/rmf_site/pull/242/commits/19483c4b7e7e83aa44ad3edfbeb37341f3919861)